File: pinyougou-manager-web/src/main/webapp/js/service/itemService.js
//服务层
app.service('itemService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../item/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../item/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../item/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../item/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../item/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../item/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../item/search.do?page='+page+"&rows="+rows, searchEntity);
	}    	
});

File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/ItemService.java
package com.pinyougou.sellergoods.service;
import java.util.List;
import com.pinyougou.pojo.TbItem;

import entity.PageResult;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface ItemService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbItem> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TbItem item);
	
	
	/**
	 * 修改
	 */
	public void update(TbItem item);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbItem findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbItem item, int pageNum,int pageSize);
	
}
